### ISSUE TITLE
Lag of keyboard input after 2 hours and 3000 logged events, constant freezing of UI

### ISSUE THREAD

**Issue by BenCallisto** [olivierfriard/BORIS#642]
**Indicate the version of BORIS**
8.20.4

**Describe the bug**
I have chained 17 videos of around 20 minutes each together in one project, all on one player in the settings to have a seamless stream. After around 2 - 2.5 hours of material playback (Full length 5 hrs 40 min.) and 3000 logged events keyboard input gets increasingly laggy. Pausing every 2-5 seconds helps to keep the delay down, but viewing around 10 seconds uninterrupted already results in several seconds of a delay to register the command. This goes for actual codes as well as simply pausing the video. I used 3000 kb/s mp4 1080p files at 30fps. Time code and frame position of the current observation is also in a lagging state, only updating to the right position after pausing for a few seconds.

This issue happened in two 4 plus hour long observations, and I have two working theories at this moment:

1) The lagging starts potentially around the mark of 3000 codes upwards.
2) In both instances I tried to run the video in x2 speed around the time lagging started. A slow down back to normal speed does not help.

Lagging doesn't seem to be file bound, after it started I jumped back to the beginning of my observation and could witness the same behavior. 

**To Reproduce**
Basically playing back 5hrs of video below 3mbit with around 3000 events logged and trying to interact with it

**Expected behavior**
I expected coding a full behavior and posture analysis for 4-5 hours spread across 17 files at a time.

**Screenshots**
Just frozen screen.

**Desktop (please complete the following information):**
 - OS: [windows 11 Pro]
 - Type of BORIS instalaltion (setup)

**In case of crash** No crash per se, but BORIS can freeze up if I try to pause or code after around 10 seconds of consecutive playing video. No crash log gets produced, and it unfreezes after a minute or two. By now it is faster and more stable to keep observing with keeping the frame by frame forward button pressed.

**Comment by olivierfriard** [comment]
Hi,
thank you for reporting this issue.
The experimental v. 8.22 should improve the BORIS behavior with a large number of coded events.
See https://github.com/olivierfriard/BORIS/releases/tag/v8.22

Please try and let me know

**Comment by BenCallisto** [comment]
> Hi, thank you for reporting this issue. The experimental v. 8.22 should improve the BORIS behavior with a large number of coded events. See https://github.com/olivierfriard/BORIS/releases/tag/v8.22
> 
> Please try and let me know

Hello Olivier,

Thank you very much, I just had a quick test run with one of my larger projects (04:56 hrs) with 7k events logged. I couldn't crash it yet, but the slow downs remain, 5 seconds playing takes another 10 - 15 seconds until it actually pauses. I will also test your improved coding maps later.

Cheers

**Comment by habana15** [comment]
Hello again,

I must say that I also still have lags after 1000 events. I think it's better and it works without lags longer than before but they still exist.

Cheers

**Comment by olivierfriard** [comment]
Hi,
Thank you for testing and reporting!
I finally found a configuration to reproduce the issue and identified some possibilities to improve the UI reactivity.
I will let you know as soon as possible (a lot of other projects...)

**Comment by olivierfriard** [comment]
Hi,
v. 8.22.1 should improve the reactivity of the GUI. See https://github.com/olivierfriard/BORIS/releases/tag/v8.22
I tested it with ~ 30K events without lags.
Please test and let me know
Cheers
